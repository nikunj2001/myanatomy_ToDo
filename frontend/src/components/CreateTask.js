import React from 'react'
import { useState } from 'react';
import {useDispatch} from "react-redux";
import { createTask } from '../store/asyncMethods/taskMethods';
const CreateTask = () => {
    const dispatch = useDispatch();
    const [state,setState] = useState({
        task:'',
        description:'',
        status:'pending'
    });
    const handleChange=e=>{
        setState({...state,[e.target.name]:e.target.value});
    }
    const submitForm=e=>{
        e.preventDefault();
       dispatch(createTask(state));
       setState({
            task:'',
        description:'',
        status:'pending'
       })
    }
    const statusChange=e=>{
        setState({...state,status:e.currentTarget.value});
    }
  return (
    <div className='form'>
        <form onSubmit={submitForm} >
           <div data-testid="todo-1" className="form__control">
            <input required value={state.task} onChange={handleChange} type="text" name='task' placeholder='Add a Task...' />    
               </div> 
               <div className="form__control">
                 <input type="text" name='description' value={state.description} onChange={handleChange} placeholder='Description' />    
                </div>
                <div  className="form__control">
                    <select name="status" value={state.status} onChange={statusChange}>
                        <option value="pending">Pending</option>    
                        <option value="under-process">Under Process</option>    
                        <option value="completed">Completed</option>    
                    </select>  
                </div>
                 <div className="form__control">
                     <input type='submit' className='btn btn-default' value='Add Task' />
                </div>         
        </form>    
    </div>
  )
}

export default CreateTask;