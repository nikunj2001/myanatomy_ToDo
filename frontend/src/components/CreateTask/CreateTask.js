import React, { useEffect } from 'react'
import { useState } from 'react';
import { useDispatch, useSelector } from "react-redux";
import { createTask } from '../../store/asyncMethods/taskMethods';
import { Toaster, toast } from "react-hot-toast"
import Store from "../../store"
const CreateTask = () => {
    const { user } = Store.getState().AuthReducer;
    const [state, setState] = useState({
        task: '',
        description: '',
        status: 'pending',
    });
    const handleChange = e => {
        setState({ ...state, [e.target.name]: e.target.value });
    }
    const submitForm = e => {
        e.preventDefault();
        if (state.task === '') {
            toast.error('Task is required!')
        } else {
            Store.dispatch(createTask(state, user._id));
            setState({
                task: '',
                description: '',
                status: 'pending',
            })
        }
    }
    const statusChange = e => {
        setState({ ...state, status: e.currentTarget.value });
    }
    return (
        <div className='form' id='create-form'>
            <form onSubmit={submitForm} >
                <Toaster
                    position="top-center"
                    reverseOrder={false}
                />
                <div data-testid="todo-1" className="form__control">
                    <input value={state.task} onChange={handleChange} type="text" name='task' placeholder='Add a Task...' />
                </div>
                <div className="form__control">
                    <input type="text" name='description' value={state.description} onChange={handleChange} placeholder='Description' />
                </div>
                <div className="form__control">
                    <select name="status" id='status' value={state.status} onChange={statusChange}>
                        <option value="pending">Pending</option>
                        <option value="under-process">Under Process</option>
                        <option value="completed">Completed</option>
                    </select>
                </div>
                <div className="form__control">
                    <input type='submit' className='btn btn-default' id='submit-btn' value='Add Task' />
                </div>
            </form>
        </div>
    )
}

export default CreateTask;