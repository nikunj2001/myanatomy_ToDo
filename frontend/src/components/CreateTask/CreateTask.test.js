import React from 'react';
import { shallow, configure } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
import CreateTask from './CreateTask';

Enzyme.configure({ adapter: new Adapter() })


describe("CreateTask", () => {

    it("should render my component", () => {
        const wrapper = shallow(<CreateTask />);
        const tree = wrapper.debug();
        expect(tree).toMatchSnapshot();
    });
});