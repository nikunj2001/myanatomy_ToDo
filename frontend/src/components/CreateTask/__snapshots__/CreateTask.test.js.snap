// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask should render my component 1`] = `
"<div className=\\"form\\" id=\\"create-form\\">
  <form onSubmit={[Function: submitForm]}>
    <Toaster position=\\"top-center\\" reverseOrder={false} />
    <div data-testid=\\"todo-1\\" className=\\"form__control\\">
      <input value=\\"\\" onChange={[Function: handleChange]} type=\\"text\\" name=\\"task\\" placeholder=\\"Add a Task...\\" />
    </div>
    <div className=\\"form__control\\">
      <input type=\\"text\\" name=\\"description\\" value=\\"\\" onChange={[Function: handleChange]} placeholder=\\"Description\\" />
    </div>
    <div className=\\"form__control\\">
      <select name=\\"status\\" id=\\"status\\" value=\\"pending\\" onChange={[Function: statusChange]}>
        <option value=\\"pending\\">
          Pending
        </option>
        <option value=\\"under-process\\">
          Under Process
        </option>
        <option value=\\"completed\\">
          Completed
        </option>
      </select>
    </div>
    <div className=\\"form__control\\">
      <input type=\\"submit\\" className=\\"btn btn-default\\" id=\\"submit-btn\\" value=\\"Add Task\\" />
    </div>
  </form>
</div>"
`;
